src/logic/community/community.ts

```diff
@@ -25,12 +26,50 @@ import {
 import { EthAddress, PaginatedParameters } from '@dcl/schemas'
 
 export function createCommunityComponent(
-  components: Pick<AppComponents, 'communitiesDb' | 'catalystClient' | 'communityRoles' | 'logs'>
+  components: Pick<AppComponents, 'communitiesDb' | 'catalystClient' | 'communityRoles' | 'logs' | 'peersStats'>
 ): ICommunityComponent {
-  const { communitiesDb, catalystClient, communityRoles, logs } = components
+  const { communitiesDb, catalystClient, communityRoles, logs, peersStats } = components
 
   const logger = logs.getLogger('community-component')
 
+  const getCommunityMembers = async (id: string, options: GetCommunityMembersOptions, userAddress?: EthAddress) => {
+    const { pagination, onlyOnline } = options
+    const communityExists = await communitiesDb.communityExists(id, { onlyPublic: !userAddress })
+
+    if (!communityExists) {
+      throw new CommunityNotFoundError(id)
+    }
+
+    const memberRole = userAddress ? await communitiesDb.getCommunityMemberRole(id, userAddress) : CommunityRole.None
+
+    if (userAddress && memberRole === CommunityRole.None) {
+      throw new NotAuthorizedError("The user doesn't have permission to get community members")
+    }
+
+    let onlinePeers: string[] | undefined = undefined
+
+    if (onlyOnline) {
+      onlinePeers = await peersStats.getConnectedPeers()
+      logger.info(`Getting community members for community using the ${onlinePeers.length} connected peers`)
+    }
+
+    const communityMembers = await communitiesDb.getCommunityMembers(id, {
+      userAddress,
+      pagination,
+      onlinePeers
```

Comment:
What about renaming this filter to something like `membersFilter` or another one that represents it is a member filter (which can be used for multiple purposes, in this case retrieving onlinePeers)